### PR TITLE
optimize push job logic

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
still seem to be experiencing push failures with this that i don't understand yet.  investigating.

**Comment by bparees** [comment]
@jhadvig ptal.  note that the push job will need to be modified to provide version numbers when this goes in.

**Comment by bparees** [comment]
@mfojtik ptal too.

**Comment by bparees** [comment]
(seems to be working now in my local tests)

**Comment by mfojtik** [comment]
@bparees this is pretty cool :-) Hopefully speeds up pushing this as now we are starting to add more versions of images... LGTM (and one kofola when you will be here)

**Comment by bparees** [comment]
:bow: thanks :)

File: lib/vagrant-openshift/action/push_openshift_images.rb [hunk 1]
```diff
@@ -70,11 +112,9 @@ def build_image(image_name, git_ref, repo_url, registry)
 git fetch && git checkout #{git_ref}
 git_ref=$(git rev-parse --short HEAD)
 echo "Building and testing #{image_name}-centos7:$git_ref ..."
-make test TARGET=centos7
-make build TARGET=centos7
+make test TARGET=centos7 VERSION=#{version}
```
Comment (jhadvig) [inline lib/vagrant-openshift/action/push_openshift_images.rb]: Aren't we going to use the `TAG_ON_SUCCESS` envar ?

File: lib/vagrant-openshift/action/push_openshift_images.rb [hunk 2]
```diff
@@ -121,20 +161,20 @@ def call(env)
           # FIXME: We always need to make sure we have the latest base image
           # FIXME: This is because the internal registry is pruned once per month
           if !@options[:build_images].include?("openshift/base")
-            @options[:build_images] = "openshift/base:master,#{@options[:build_images]}"
+            @options[:build_images] = "openshift/base:1:master,#{@options[:build_images]}"
```
Comment (jhadvig) [inline lib/vagrant-openshift/action/push_openshift_images.rb]: What kind of black magic is this :flushed: ?

Comment (bparees) [inline lib/vagrant-openshift/action/push_openshift_images.rb]: we now require the version to build as part of the field, and for sti-base we just default to "1"

File: lib/vagrant-openshift/action/push_openshift_images.rb (hunk 1 above)
Comment (bparees) [inline lib/vagrant-openshift/action/push_openshift_images.rb]: ah yes sorry i was thinking that was default behavior.  why isn't it default behavior?

**Pull request merged**